### DISCUSSION TITLE
Disabling Transfer-Encoding: chunked on production.

### DISCUSSION THREAD

**Discussion by deepsyx** [vercel/next.js#38606] [Help]
Hello,
We've been using NextJS 12.1.0 with Cloudfront successfully for some time. So far the next compress was set to false and we relied on AWS Cloudfront to compress the files (brotli/gzip).

Recently we migrated to 12.2.0 and the HTTP headers from the next were changed to follow the HTTP/1.1 standard instead of the HTTP/1.0 - before the migration we received the Content-Length header (which is required for Cloudfront to work).

> - The file must be of a type that CloudFront compresses.
- The file size must be between 1,000 and 10,000,000 bytes.
- The response must include a Content-Length header so CloudFront can determine whether the file size is in the range that CloudFront compresses. If the Content-Length header is missing, CloudFront won't compress the file.
- The response must not include a Content-Encoding header.

However, after we migrated to 12.2.0 all requests followed the HTTP/1.1 protocol meaning that we no longer receive Content-Length header, but we started receiving `Transfer-Encoding: chunked` and the files were sent in chunks. 

Is there a way to bring back the old behavior without using additional reverse proxies or if not - what would be the proper solution to our problem? We'd like to not compress the files at NextJS level and only compress them at our reverse proxy.

**Comment by mderazon** [comment]
Curious to know if you were able to resolve this issue ?

**Reply by esterfania** [reply]
Me neither

**Reply by Lemonade4400** [reply]
I think I'm having the same issue...

**Reply by KirillSuhodolov** [reply]
same issue

**Reply by smaeda-ks** [reply]
FYI, [Disable streaming for `pages`](https://github.com/vercel/next.js/pull/45895) has landed in:
https://github.com/vercel/next.js/releases/tag/v13.2.0

**Reply by tomitrescak** [reply]
@smaeda-ks but is there any flag for it ? I am on Next 13.4.4 and I still recieve all requests in chunked encoding in dev

**Comment by mderazon** [comment]
Related ?
https://github.com/vercel/next.js/pull/45895

**Comment by tamebadger** [comment]
Would be cool to get an answer here, running into similar problems :(

**Comment by dnhuan** [comment]
Running into the same issue here. `vc dev` does not return chunked data, but the production server does

**Comment by G07cha** [comment]
I was having this issue in dev on Next 14 and found the solution in openapi-typescript discussions - https://github.com/openapi-ts/openapi-typescript/discussions/1912#discussioncomment-11226631
